eddymotion/model.py

```diff
@@ -53,15 +53,17 @@ def init(gtab, model="DTI", **kwargs):
                 "lambdaL": 1e-8,
             }
 
-        elif model.lower().startswith("sfm"):
-            from eddymotion.utils.model import (
-                SFM4HMC as Model,
-                ExponentialIsotropicModel,
-            )
+        elif model.lower() in ("sfm", "gp"):
+            Model = SparseFascicleModel
+            if model.lower() == "sfm":
+                param = {
+                    "solver": "ElasticNet",
+                    "isotropic": ExponentialIsotropicModel,
+                }
+            else:
+                from sklearn.gaussian_process import GaussianProcessRegressor
 
-            param = {
-                "isotropic": ExponentialIsotropicModel,
-            }
+                param = {"solver": GaussianProcessRegressor}
```

Comment:
In the long-term, I'd bet we get more miles out of using https://docs.pymc.io/en/v3/Gaussian_Processes.html in place of sklearn's GaussianProcessRegressor but for now this seems fine!

RE: tuning this, we could start by trying to make alpha very small like what we did with the elastic-net case, but I don't know about the kernel hyperparameters. @arokem -- would the idea here be that by wrapping this in SFM, we'd avoid the need for learning the shape of a manual kernel? The default used for GP in sklearn is `ConstantKernel(1.0, constant_value_bounds="fixed" * RBF(1.0, length_scale_bounds="fixed")`.

If we're sticking with this, then I think that either way, we should remove the `fixed` option since "kernel hyperparameters are optimized during fitting unless the bounds are marked as `fixed`" and from Andersson & Sotiropoulos 2015: "The smoothness is determined from hyperparameters whose values are determined directly from the data" and "the hyperparameters will be such that the predictions are a smooth function of gradient direction and, in the case of multi-shell data, b-value." Also noteworthy was this point: "the data given by original hyperparameters is very sharp and even for a dataset with 300 points will give almost half the weight to the center-point (i.e., when predicting the signal for a diffusion direction half the information comes from the point itself)."

Reply:
And for selecting the kernel, https://www.sciencedirect.com/science/article/pii/S1053811915006874?via%3Dihub#:~:text=Finding%20k(x,scaling%20or%20variance. they suggest using either exponential or spherical and then optimizing with maximum likelihood